### PR TITLE
Enable adding/removing cross space/org network policies

### PR DESCRIPTION
**Pull request by christianang**
## Does this PR modify CLI v6 or v7?

v6

## What Need Does It Address?

Users would like to be able to add/remove the network policies.

## Description of the Change

We've added `-o` and `-s` flags to the `cf add-network-policy` and `cf remove-network-policy` command and updated the help text. This change brings the CLI command in line with the API.

## How Urgent Is The Change?

Not.

## Other Relevant Parties

@jsirlin @pvaramballypivot
@oozie @XanderStrike @ameowlia

https://www.pivotaltracker.com/story/show/161939743
https://www.pivotaltracker.com/story/show/161969089

### CODE REVIEW

**Comment by christianang** [comment]
Updated this PR based on feedback from https://github.com/cloudfoundry/cli/pull/1504.

File: command/v6/add_network_policy_command_test.go [hunk 1]
```diff
@@ -150,11 +160,99 @@ var _ = Describe("add-network-policy Command", func() {
...
+				Expect(executeErr).To(MatchError(translatableerror.NetworkPolicyDestinationOrgWithoutSpaceError{}))
+				Expect(testUI.Out).NotTo(Say(`Adding network policy`))
+			})
+		})
+
+		When("invalid org is specified", func() {
+			BeforeEach(func() {
+				cmd.DestinationOrg = "bananarama"
+				cmd.DestinationSpace = "hamdinger"
+				fakeActor.GetOrganizationByNameReturns(v3action.Organization{}, v3action.Warnings{}, actionerror.OrganizationNotFoundError{Name: "bananarama"})
```
Comment (tjvman) [inline command/v6/add_network_policy_command_test.go]: Please add warnings here and corresponding assertions testing that those warnings are printed.

File: command/v6/add_network_policy_command_test.go [hunk 2]
```diff
@@ -150,11 +160,99 @@ var _ = Describe("add-network-policy Command", func() {
...
+				Expect(executeErr).To(MatchError(actionerror.OrganizationNotFoundError{Name: "bananarama"}))
+				Expect(testUI.Out).NotTo(Say(`Adding network policy`))
+			})
+		})
+
+		When("a valid org but invalid space is specified", func() {
+			BeforeEach(func() {
+				cmd.DestinationOrg = "bananarama"
+				cmd.DestinationSpace = "hamdinger"
+				fakeActor.GetOrganizationByNameReturns(v3action.Organization{GUID: "some-org-guid"}, v3action.Warnings{}, nil)
```
Comment (tjvman) [inline command/v6/add_network_policy_command_test.go]: Please add warnings here and on the line below and corresponding assertions testing that those warnings are printed.

File: command/v6/add_network_policy_command_test.go [hunk 3]
```diff
@@ -150,11 +160,99 @@ var _ = Describe("add-network-policy Command", func() {
...
+				Expect(passedOrgGuid).To(Equal("some-org-guid"))
+				Expect(executeErr).To(MatchError(actionerror.SpaceNotFoundError{Name: "bananarama"}))
+				Expect(testUI.Out).NotTo(Say(`Adding network policy`))
+			})
+		})
+
+		When("a destination space but no destination org is specified", func() {
+			BeforeEach(func() {
+				cmd.DestinationSpace = "hamdinger"
+				fakeActor.GetSpaceByNameAndOrganizationReturns(v3action.Space{GUID: "some-other-space-guid"}, v3action.Warnings{}, nil)
```
Comment (tjvman) [inline command/v6/add_network_policy_command_test.go]: Please add warnings here and corresponding assertions testing that those warnings are printed.

File: command/v6/remove_network_policy_command.go [hunk 4]
```diff
@@ -52,11 +58,42 @@ func (cmd *RemoveNetworkPolicyCommand) Setup(config command.Config, ui command.U
 }
 
 func (cmd RemoveNetworkPolicyCommand) Execute(args []string) error {
+	switch {
```
Comment (tjvman) [inline command/v6/remove_network_policy_command.go]: A `switch` statement feels odd here, given that there's only one case.

File: command/v6/remove_network_policy_command_test.go [hunk 5]
```diff
@@ -106,17 +138,82 @@ var _ = Describe("remove-network-policy Command", func() {
...
+			It("responds with an error", func() {
+				Expect(executeErr).To(MatchError(translatableerror.NetworkPolicyDestinationOrgWithoutSpaceError{}))
+			})
+		})
+
+		When("an invalid org is provided", func() {
+			BeforeEach(func() {
+				cmd.DestinationOrg = "coolorg"
+				cmd.DestinationSpace = "coolspace"
+				fakeActor.GetOrganizationByNameReturns(v3action.Organization{}, v3action.Warnings{}, actionerror.OrganizationNotFoundError{Name: "coolorg"})
```
Comment (tjvman) [inline command/v6/remove_network_policy_command_test.go]: Please add warnings here and corresponding assertions testing that those warnings are printed.

File: command/v6/remove_network_policy_command_test.go [hunk 6]
```diff
@@ -106,17 +138,82 @@ var _ = Describe("remove-network-policy Command", func() {
...
+				Expect(executeErr).To(MatchError(actionerror.OrganizationNotFoundError{Name: "coolorg"}))
+			})
+		})
+
+		When("an invalid space is provided", func() {
+			BeforeEach(func() {
+				cmd.DestinationOrg = "coolorg"
+				cmd.DestinationSpace = "coolspace"
+				fakeActor.GetOrganizationByNameReturns(v3action.Organization{GUID: "some-org-guid"}, v3action.Warnings{}, nil)
+				fakeActor.GetSpaceByNameAndOrganizationReturns(v3action.Space{}, v3action.Warnings{}, actionerror.SpaceNotFoundError{Name: "coolspace"})
```
Comment (tjvman) [inline command/v6/remove_network_policy_command_test.go]: Please add warnings here and corresponding assertions testing that those warnings are printed.

File: actor/cfnetworkingaction/policy.go [hunk 7]
```diff
@@ -178,3 +179,11 @@ func (Actor) transformPolicy(appNameByGuid map[string]string, v1Policy cfnetv1.P
 	}
 	return Policy{}
 }
+
```
[the remark this answers is not in the document]
Comment (christianang) [reply actor/cfnetworkingaction/policy.go]: addressed

File: command/v6/add_network_policy_command_test.go [hunk 8]
```diff
@@ -150,11 +160,99 @@ var _ = Describe("add-network-policy Command", func() {
...
+
+		When("org is specified and space is not", func() {
+			BeforeEach(func() {
+				cmd.DestinationOrg = "bananarama"
+				cmd.DestinationSpace = ""
+			})
+
+			It("returns an error", func() {
+				Expect(executeErr).To(MatchError(translatableerror.NetworkPolicyDestinationOrgWithoutSpaceError{}))
+				Expect(testUI.Out).NotTo(Say(`Adding network policy`))
```
[the remark this answers is not in the document]
Comment (christianang) [reply command/v6/add_network_policy_command_test.go]: done

File: command/v6/add_network_policy_command_test.go [hunk 9]
```diff
@@ -150,11 +160,99 @@ var _ = Describe("add-network-policy Command", func() {
...
+		When("invalid org is specified", func() {
+			BeforeEach(func() {
+				cmd.DestinationOrg = "bananarama"
+				cmd.DestinationSpace = "hamdinger"
+				fakeActor.GetOrganizationByNameReturns(v3action.Organization{}, v3action.Warnings{}, actionerror.OrganizationNotFoundError{Name: "bananarama"})
+			})
+
+			It("returns an error", func() {
+				Expect(executeErr).To(MatchError(actionerror.OrganizationNotFoundError{Name: "bananarama"}))
+				Expect(testUI.Out).NotTo(Say(`Adding network policy`))
```
[the remark this answers is not in the document]
Comment (christianang) [reply command/v6/add_network_policy_command_test.go]: good idea

File: command/v6/add_network_policy_command_test.go [hunk 10]
```diff
@@ -150,11 +160,99 @@ var _ = Describe("add-network-policy Command", func() {
...
+				fakeActor.GetOrganizationByNameReturns(v3action.Organization{GUID: "some-org-guid"}, v3action.Warnings{}, nil)
+				fakeActor.GetSpaceByNameAndOrganizationReturns(v3action.Space{}, v3action.Warnings{}, actionerror.SpaceNotFoundError{Name: "bananarama"})
+			})
+
+			It("returns an error", func() {
+				passedSpaceName, passedOrgGuid := fakeActor.GetSpaceByNameAndOrganizationArgsForCall(0)
+				Expect(passedSpaceName).To(Equal("hamdinger"))
+				Expect(passedOrgGuid).To(Equal("some-org-guid"))
+				Expect(executeErr).To(MatchError(actionerror.SpaceNotFoundError{Name: "bananarama"}))
+				Expect(testUI.Out).NotTo(Say(`Adding network policy`))
```
[the remark this answers is not in the document]
Comment (christianang) [reply command/v6/add_network_policy_command_test.go]: 💯

File: command/v6/add_network_policy_command_test.go (hunk 1 above)
Comment (christianang) [reply command/v6/add_network_policy_command_test.go]: 👍

File: command/v6/add_network_policy_command_test.go (hunk 2 above)
Comment (christianang) [reply command/v6/add_network_policy_command_test.go]: 🥇

File: command/v6/add_network_policy_command_test.go (hunk 3 above)
Comment (christianang) [reply command/v6/add_network_policy_command_test.go]: 😸

File: command/v6/remove_network_policy_command.go (hunk 4 above)
Comment (christianang) [reply command/v6/remove_network_policy_command.go]: 🤔

File: command/v6/remove_network_policy_command_test.go (hunk 5 above)
Comment (ameowlia) [reply command/v6/remove_network_policy_command_test.go]: done

File: command/v6/remove_network_policy_command_test.go (hunk 6 above)
Comment (ameowlia) [reply command/v6/remove_network_policy_command_test.go]: 👾

**Comment by ameowlia** [comment]
We are still making updates and will ping you all when we are ready for you all to merge 🐱

**Comment by ameowlia** [comment]
We finished our updates and are ready for you all to review again! 

Thanks, 
Amelia and @christianang

**Comment by ameowlia** [comment]
Thanks for helping us with this @tjvman, @jenspinney, @abbyachau!! 🎉 🎉 🎉

**Pull request merged**